scripts/common/image_repository_helpers.py

```diff
@@ -173,16 +206,139 @@ def from_artifact_repository_image(image_data: DockerImage) -> 'Image':
         )
 
 
+@dataclass
+class Repository:
+    images: list[Image]
+
+    def includes_image_version(self, image: Image):
+        """
+        Check if this repository already has a certain version of an image
+        """
+        matching_images = [
+            i for i in self.images if i.name == image.name and i.digest == image.digest
+        ]
+        return len(matching_images) > 0
```

Comment:
Bit trivial, but these four lines seem like a job for `any(…)`.

Reply:
Good point, have updated